01_fib/cuminandpaprika/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+// Print n number of fibonnaci sequence
+// 1, 1, 2, 3, 5, 8, 13
+// Note: we take the absolute value of n to ensure our loop works.
+func fib(n int16) {
+	var i int16
+	if isPositive(n) {
+		for ; i < n; i++ {
+			fmt.Fprintln(out, calcFib(i))
+		}
+		return
+	}
+
+	for ; i > n; i-- {
+		fmt.Fprintln(out, calcFib(i))
+	}
+
+}
+
+func calcFib(n int16) int16 {
+	if isPositive(n) {
+		return calcPositiveFib(n)
+	}
+	return calcNegativeFib(n)
+}
+
+func calcPositiveFib(n int16) int16 {
+	if n == 0 {
```

Comment:
appreciate the intent here, so again this is a nitpick and potentially personal taste:

I'd be inclined to put this early out higher up in the call stack, as it has knock on effects:

- your `isPositive` is checking for `0` as well simply to make this work
- in order to see this particular branch we're already three functions deep, so from a readability perspective it's less transparent.  Obviously not a problem in such a short program, but if it were more complicated then moving the early out higher up the tree can make it easier to reason about the program as a whole

Reply:
I definitely see your point here but my intent was to keep calcPositiveFib a relatively generic function that can be reused by functions other than fib. 

I've made a change to handle it earlier but I find myself still needing to handle the zero case due to the recursion. Perhaps it's time to use a loop rather than incursion... 

What are your thoughts?